source/plugins/ruby/ApplicationInsightsUtility.rb

```diff
@@ -323,5 +325,46 @@ def getWorkspaceCloud()
         $log.warn("Exception in AppInsightsUtility: getWorkspaceCloud - error: #{errorStr}")
       end
     end
+
+    def sendAPIResponseTelemetry(responseCode, resource, metricName)
+      begin
+        if (!responseCode.nil? && !responseCode.empty?)
+          if (@@apiResponseCodeHash.has_key?(responseCode))
```

Comment:
this should be **not** of the has_key, right?
!@@apiResponseCodeHash.has_key?(responseCode)